library/core/src/time.rs

```diff
@@ -208,6 +222,9 @@ impl Duration {
     #[inline]
     #[must_use]
     #[rustc_const_stable(feature = "duration_consts_2", since = "1.58.0")]
+    // by definition of NANOS_PER_SEC, the checks for div by 0 cases are unreachable, but also unneeded
+    #[requires(nanos < NANOS_PER_SEC || secs.checked_add((nanos / NANOS_PER_SEC) as u64).is_some())]
```

Comment:
This is a `safe` function, why do we have a `requires` clause?

Reply:
The function is not marked `unsafe`, which implies that it must not exhibit undefined behaviour for any input (per Rust's compile-time guarantees). A `panic` isn't undefined behaviour, so there isn't a need to guarantee that this does not happen. We may, however, still need to tweak something with regard to the `should_fail` predicate of Kani to make sure Kani isn't unhappy.

Reply:
We can remove the `kani::assume` from `Duration::any` because we don't need the assumption if we've removed the precondition. Then we can add a `should_panic` to the contract proof for `Duration::new()`.  Like so: 

```rust
impl kani::Arbitrary for Duration {
        fn any() -> Duration {
            let secs = kani::any::<u64>();
            let nanos = kani::any::<u32>();
            //kani::assume(nanos < NANOS_PER_SEC || secs.checked_add((nanos / NANOS_PER_SEC) as u64).is_some());
            Duration::new(secs, nanos)
        }
    }

    #[kani::proof_for_contract(Duration::new)]
    #[kani::should_panic]
    fn duration_new() {
        let secs = kani::any::<u64>();
        let nanos = kani::any::<u32>();
        let _ = Duration::new(secs, nanos);
    }
```

Kani is happy with this contract, but now the following contracts fail because a call to `Duration::any()` can now panic: 
```
Verification failed for - time::duration_verify::duration_checked_mul
Verification failed for - time::duration_verify::duration_checked_sub
Verification failed for - time::duration_verify::duration_checked_add
Verification failed for - time::duration_verify::duration_as_nanos
Verification failed for - time::duration_verify::duration_as_micros
Verification failed for - time::duration_verify::duration_as_millis
Verification failed for - time::duration_verify::duration_subsec_nanos
Verification failed for - time::duration_verify::duration_subsec_micros
Verification failed for - time::duration_verify::duration_subsec_millis
Verification failed for - time::duration_verify::duration_as_secs
```

We can add `should_panic` to all of these contract proofs, but I'm not sure if that's the best option.